test/integration/utils/certificates.go

```diff
@@ -140,5 +140,9 @@ func GenerateCertificates(config *CertificatesConfig) error {
 		return err
 	}
 	pem.Encode(keyOut, p)
-	return nil
+	err = certOut.Sync()
+	if cerr := keyOut.Sync(); err == nil {
+		err = cerr
```

Comment:
here maybe you ca concatenate the 2 errors to keep track of the first error. for that you can use the [`errors` package and the `Join` function](https://pkg.go.dev/errors#Join)

```suggestion
		err = errors.Join(err, cerr)
```

Reply:
`errors.Join()` was introduced in errors v1.20 but we are using v0.9.1.
Is there a way to combine these two errors without a join ?

Reply:
yes we use another package already in the agent called: k8s.io/apimachinery/pkg/util/errors  and the [NewAggregate ](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/errors#NewAggregate) function.

but it is less convenient that `Join`